flocker/provision/_install.py

```diff
@@ -44,9 +44,9 @@
                     archive_bucket=ARCHIVE_BUCKET,
                     ),
     # FLOC-1828 TODO - use ubuntu rather than ubuntu-testing
-    'ubuntu-14.04': 'https://{archive_bucket}.s3.amazonaws.com/'
-                'ubuntu-testing/14.04/$(ARCH)'.format(
-                    archive_bucket=ARCHIVE_BUCKET
+    'ubuntu-14.04': 'https://{archive_bucket}.s3.amazonaws.com/ubuntu-testing/'
+                    '$(lsb_release --release --short)/\\$(ARCH)'.format(
```

Comment:
This value is evaluated as a shell command. The ` `$(lsb...)` part must be substituted as part of that command evaluation.  The `$(ARCH)` part must be left unevaluated, hence the backslash escapes (one to make shell ignore the `$` as a substitution marker, and then doubled to make Python ignore the `\` as an escape marker).

The output of this value then goes into `/etc/apt/sources.list` which does its own substitution on `$(ARCH)` during a subsequent `apt-get update`.